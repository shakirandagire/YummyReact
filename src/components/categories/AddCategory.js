import React, { Component } from 'react';
import axios from 'axios';
import { notify } from 'react-notify-toast';
import { BASE_URL } from '../../../src/constants';
import Navigation from '../other/Navigation';

/**
 * Component for  adding categories
 */
class AddCategory extends Component {
    state = {
      categoryname: '',
      category_description: '',
    }

      handleInputChange = (event) => {
        const { name, value } = event.target;
        this.setState({ [name]: value });
      }
      // Method that handles the adding of categories
      handleAddCategory = (event) => {
        const headers = { Authorization: `Bearer ${localStorage.getItem('accessToken')}` };
        event.preventDefault();
        axios
          .post(`${BASE_URL}/api/v1/categories/`, this.state, { headers })
          .then((response) => {
            notify.show(response.data.message, 'success', 1000);
            this.props.history.push('/view-categories');
          })
          .catch((error) => {
            if (error.response) {
              notify.show(error.response.data.message);
            } else if (error.request) {
              notify.show("Request not made");
            }
          });
      }

      render() {
        return (
          <div className="my_container backgimg">
            <Navigation />
            <div className="mycontent">

              <form onSubmit={this.handleAddCategory}>

                <div className="form-group">
                  Category Name:
                  <input type="text" name="categoryname" className="form-control categoryname" required onChange={this.handleInputChange} value={this.state.categoryname} />
                </div>
                <div className="form-group">
                  Description:
                  <input type="text" name="category_description" className="form-control category_description" required onChange={this.handleInputChange} value={this.state.category_description} />
                </div>

                <button type="submit" className="btn btn-danger formsave" >Save</button>
              </form>
            </div>

          </div>


        );
      }
}
export default AddCategory;
