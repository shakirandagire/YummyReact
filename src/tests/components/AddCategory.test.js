import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import AddCategory from '../../components/AddCategory';

const component = shallow(<AddCategory />);
describe('<AddCategory/>', () => {
  it('should render properly', () => {
    expect(component.length).toBe(1);
  });

  it('should render view categories', () => {
    expect(shallowToJson(component)).toMatchSnapshot();
  });
});

