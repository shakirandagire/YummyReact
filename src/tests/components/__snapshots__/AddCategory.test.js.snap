// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddCategory/> should render add categories 1`] = `
<div
  className="my_container backgimg"
>
  <Navigation />
  <div
    className="mycontent"
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        Category Name:
        <input
          className="form-control"
          name="categoryname"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        Description:
        <input
          className="form-control"
          name="category_description"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
      </div>
      <button
        className="btn btn-danger formsave"
        type="submit"
      >
        Save
      </button>
    </form>
  </div>
</div>
`;
